hydrolib/core/io/bc/models.py

```diff
@@ -220,23 +271,177 @@ class T3D(ForcingBase):
     function: Literal["t3d"] = "t3d"
 
     offset: float = Field(0.0, alias="offset")
+    """float: All values in the table are increased by the offset (after multiplication by factor). Defaults to 0.0."""
+
     factor: float = Field(1.0, alias="factor")
+    """float: All values in the table are multiplied with the factor. Defaults to 1.0."""
 
-    verticalpositions: List[float] = Field(alias="verticalPositions")
-    verticalinterpolation: VerticalInterpolation = Field(alias="verticalInterpolation")
-    verticalpositiontype: VerticalPositionType = Field(alias="verticalPositionType")
+    vertpositions: List[float] = Field(alias="vertPositions")
+    """List[float]: The specification of the vertical positions."""
+
+    vertinterpolation: VerticalInterpolation = Field(
+        VerticalInterpolation.linear.value, alias="vertInterpolation"
+    )
+    """VerticalInterpolation: The type of vertical interpolation. Defaults to linear."""
+
+    vertpositiontype: VerticalPositionType = Field(alias="vertPositionType")
+    """VerticalPositionType: The vertical position type of the verticalpositions values."""
+
+    timeinterpolation: TimeInterpolation = Field(
+        TimeInterpolation.linear.value, alias="timeInterpolation"
+    )
+    """TimeInterpolation: The type of time interpolation. Defaults to linear."""
+
+    _key_renaming_root_validator = get_key_renaming_root_validator(
+        {
+            "timeinterpolation": ["time_interpolation"],
+            "vertpositions": ["vertical_position_specification"],
```

Comment:
I don't really get these keywords?
To be backwards compatible, shouldn't for examle  `verticalpositions` be part of the old keys, since that was our old key?
Also are spaces replaced with underscores? Is that the reason why it is `vertical_position_specification` instead of `vertical position specification`?

Reply:
The idea here is that the first keyword (i.e the key of the dictionary) is the current keyword that should be used. What follows (the value belonging to a key) is a list of older keywords. So we can look for multiple older keywords and rename them to the current version of the keyword. Do you have any idea how we can make this more clear?

I will ask @arthurvd if we should add more old keys here, like the one you mentioned.

Regarding your question about spaces and underscores: Indeed, the keyword specified in the file can be `Vertical Position Specification`, for example. At first I thought pydantic replaced the spaces with underscores, but after a little more digging in our code, it turns out we convert those spaces to underscores in our `to_key` function that constructs a key name from a given field name. So that's why I had to specify the keyword with underscores.

Do you think it would be more clear if we used `to_key("Vertical Position Specification")` here instead?

Reply:
Discussed this with Prisca. We will not use the `to_key` function here.